src/sentry/tasks/unmerge.py

```diff
@@ -273,6 +280,27 @@ def truncate_denormalizations(group):
     features.delete(group)
 
 
+def collect_group_environment_data(events):
+    results = {}
+    for event in events:
+        results[(event.group_id, get_environment_name(event))] = event.get_tag('sentry:release')
```

Comment:
Is it never the case that there are 2 events with the same `(event.group_id, get_environment_name(event))` but different releases? Should one "win"?

Reply:
This is trying to find the first release for a `(group, environment)` pair from a list of date-descending sorted events, so yeah, the last-update-wins approach is intended here. This would probably be good to comment because definitely can see in retrospect how this is unclear.